plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1446,23 +1446,38 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
             throw new TrinoException(NOT_SUPPORTED, "This connector does not support adding not null columns");
         }
         Table icebergTable = catalog.loadTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName());
-        icebergTable.updateSchema().addColumn(column.getName(), toIcebergType(column.getType()), column.getComment()).commit();
+        try {
+            icebergTable.updateSchema().addColumn(column.getName(), toIcebergType(column.getType()), column.getComment()).commit();
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, e);
```

Comment:
Do add also a summary of the failure to the `TrinoException` - e.g. `"Failed to add column %s".formatted(column.getName())`

Reply:
I initially had that, but some tests actually expect the message of the underlying exception

Reply:
> some tests actually expect the message of the underlying exception

Maybe we should adapt them then. 
I think the exceptions messages shown to the users should be meaningful and describe high level the problem which occurred.

Reply:
I adjusted the affected tests @findinpath